components/script/dom/bindings/typedarrays.rs

```diff
@@ -9,30 +9,36 @@ use std::ptr;
 use js::jsapi::{Heap, JSObject, JS_GetArrayBufferViewBuffer};
 use js::rust::wrappers::DetachArrayBuffer;
 use js::rust::{CustomAutoRooterGuard, MutableHandleObject};
-use js::typedarray::{
-    CreateWith, Float32Array, JSObjectStorage, TypedArray, TypedArrayElement,
-    TypedArrayElementCreator,
-};
+use js::typedarray::{CreateWith, TypedArray, TypedArrayElement, TypedArrayElementCreator};
 
 use crate::script_runtime::JSContext;
 
 #[derive(Default, JSTraceable)]
-pub struct HeapFloat32Array {
+pub struct HeapTypedArray {
```

Comment:
@gterzian please review, I tried to make it HeapTypedArray<T> but since T is not used here I added `PhantomData ` but I got errors because of `JSTraceable `

Reply:
on phantom field you need to add `#[no_trace]`